### ISSUE TITLE
[WAGON-467] wagon-ssh: options to configure hostKeyChecking and PreferredAuthentications 

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-wagon#536]
**[Nikolas Falco](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nfalco79)** opened **[WAGON-467](https://issues.apache.org/jira/browse/WAGON-467?redirect=false)** and commented

Hi,
I'm looking a way to deploy the maven site using scp protocol.
I found some issues:
* when I create a Jenkins node the maven process stucks during site:deploy goal because it prompts to me if I want add the destination host key to know_hosts file.
* also with correct credential in settings.xml it prompts to me for Kerberos username [$USER] and password.

After a lot of google search the SSH options command to setup should be:
* StrictHostKeyChecking [ask, no, yes]
* PreferredAuthentications a mix of [gssapi-with-mic, publickey, password, keyboard-interactive]

Configure those option into them into .ssh/config file it's hard to manage in many in case of many machine

Actually there is not way to change those settings in the wagon instance
For StrictHostKeyChecking could be used the settings.xml

```xml
<servers>
	<server>
		<id>docserver</id>
		<username>xxx</username>
		<password>password</password>
		<configuration>
			<knownHostsProvider>
				<hostKeyChecking>no</hostKeyChecking>
			</knownHostsProvider>
		</configuration>
	</server>
</servers>
```

but it fails because KnownHostsProvider it's an interface and if I specify the implementation of knownHostsProvider

```xml
<knownHostsProvider implementation="org.apache.maven.wagon.providers.ssh.knownhost.FileKnownHostsProvider">
```

fails because ClassNotFound on IOUtil in super constructor method.

Instead for PreferredAuthentications it's an plain string into the code so no way to change it.

---

**Affects:** 2.10

**Issue Links:**
- [WAGON-355](https://issues.apache.org/jira/browse/WAGON-355) Expose PreferredAuthentications property of jsch in some way
 (_**"is duplicated by"**_)
- [WAGON-323](https://issues.apache.org/jira/browse/WAGON-323) -keyboard-interactive method is used even when password is supplied

- [WAGON-415](https://issues.apache.org/jira/browse/WAGON-415) wagon-ssh doesn't load known_hosts from Jenkins home directory

- [WAGON-454](https://issues.apache.org/jira/browse/WAGON-454) Settings.xml server configuration not inyected to wagon extension

- [WAGON-439](https://issues.apache.org/jira/browse/WAGON-439) Jsch issue in java 1.7 and Kerberos-enabled ssh servers
 (_**"is superceded by"**_)

**Remote Links:**
- [GitHub Pull Request #29
](https://github.com/apache/maven-wagon/pull/29)

**Comment by jira-importer** [comment]
**[Dan Tran](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dantran)** commented

Hi Nikolas, with your patch, are you able to customize wagon ssh using maven's settings.xml?

**Comment by jira-importer** [comment]
**[Nikolas Falco](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nfalco79)** commented

Yes. Follow an example of my settings.xml that upload the maven site of our multi module project

```xml
<servers>
  <server>
    <id>docserver</id>
    <username>user</username>
    <password>password</password>
    <configuration>
      <strictHostKeyChecking>no</strictHostKeyChecking>
      <preferredAuthentications>publickey,password</preferredAuthentications>
      <interactive>false</interactive>
    </configuration>
  </server>
</servers>
```

The interactive element is not needed, anyway I set it to false (default is true).
With above settings it doesn't prompt in case of new host keys to add and for kerberos username and password.
Correct values for preferredAuthentications I've got [here](http://stackoverflow.com/questions/29669459/skipping-kerberos-authentication-prompts-with-jsch)
